File: lab/cometbft/shared/blockchain_burst.js
const http = require('http');

const n = parseInt(process.argv[2]);
const ips = process.argv.slice(3);

if (isNaN(n) || ips.length === 0) {
    console.log("Usage: node blockchain_burst.js <N> [node_ips...]");
    process.exit(1);
}

console.log(`--- Blockchain Direct Burst START (N=${n}) ---`);
console.log(`Targets: ${ips.join(', ')}`);

const start = Date.now();
let completed = 0;
let errors = 0;

const sendRequest = (ip) => {
    return new Promise((resolve) => {
        const data = JSON.stringify({ type: "SQL_INJECTION", value: 1 });
        const options = {
            hostname: ip,
            port: 3000,
            path: '/stress',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': data.length
            },
            timeout: 10000
        };

        const req = http.request(options, (res) => {
            res.on('data', () => {});
            res.on('end', () => resolve(true));
        });

        req.on('error', () => resolve(false));
        req.on('timeout', () => {
            req.destroy();
            resolve(false);
        });
        
        req.write(data);
        req.end();
    });
};

async function burst() {
    const promises = [];
    for (let i = 0; i < n; i++) {
        const ip = ips[i % ips.length];
        promises.push(sendRequest(ip).then(success => {
            completed++;
            if (!success) errors++;
        }));
    }
    await Promise.all(promises);
    const duration = (Date.now() - start) / 1000;
    console.log(`--- Blockchain Direct Burst END (Duration: ${duration.toFixed(2)}s, Errors: ${errors}) ---`);
}

burst();

File: lab/cometbft/shared/blockchain_burst.py
import sys
import time
import random
import urllib.request
import json
from concurrent.futures import ThreadPoolExecutor

def send_request(idx, target_ip):
    url = f"http://{target_ip}:3000/stress"
    # Send empty or generic data
    data = json.dumps({"type": "SQL_INJECTION", "value": 1}).encode('utf-8')
    try:
        req = urllib.request.Request(url, data=data, method="POST", headers={'Content-Type': 'application/json'})
        with urllib.request.urlopen(req, timeout=10) as response:
            pass 
    except Exception:
        pass

def main():
    if len(sys.argv) < 2:
        print("Usage: python3 blockchain_burst.py <N> [node_ips...]")
        sys.exit(1)
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Error: N must be an integer")
        sys.exit(1)
        
    ips = sys.argv[2:]
    if not ips:
        print("Error: No target IPs provided")
        sys.exit(1)
    
    print(f"--- Blockchain Direct Burst START (N={n}) ---")
    print(f"Targets: {ips}")
    
    start_time = time.time()
    # High concurrency for burst density
    max_workers = min(200, n)
    with ThreadPoolExecutor(max_workers=max_workers) as executor:
        for i in range(n):
            target = ips[i % len(ips)]
            executor.submit(send_request, i, target)
            
    duration = time.time() - start_time
    print(f"--- Blockchain Direct Burst END (Duration: {duration:.2f}s) ---")

if __name__ == "__main__":
    main()
